### PR TITLE
feat(iam): add a new data source to query permissions

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

Add a new data source: **huaweicloud_identity_permissions**

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/iam" TESTARGS="-run TestAccIdentityPermissionsDataSource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityPermissionsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityPermissionsDataSource_basic
=== PAUSE TestAccIdentityPermissionsDataSource_basic
=== CONT  TestAccIdentityPermissionsDataSource_basic
--- PASS: TestAccIdentityPermissionsDataSource_basic (12.60s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       12.647s
```

### CODE REVIEW

File: huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	}
+	return &sc
+}
+
+// dataSourceIdentityPermissionsRead performs the permissions lookup
+func dataSourceIdentityPermissionsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	identityClient, err := cfg.IdentityV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("Error creating identity client: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go]:
```suggestion
		return diag.Errorf("error creating identity client: %s", err)
```

File: huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+	listOpts := roles.ListOpts{
+		DisplayName: d.Get("name").(string),
+		Catalog:     d.Get("catalog").(string),
+	}
+
+	roleType := d.Get("type").(string)
+	if roleType == "custom" {
+		if cfg.DomainID == "" {
+			return diag.Errorf("the domain_id must be specified in the provider configuration")
```
Comment (niuzhenguo) [inline huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go:97]: `domain_id` is not displayed on our docs, better to use `domain_name` here.

Comment (ShiChangkuo) [reply huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go:97]: It is only happens in the internal environment.
The `domain_id` parameter will make the provider don't call the API which is not exist in the internal environment.

Comment (niuzhenguo) [reply huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go:97]: ok

**Pull request merged**